app/src/main/java/com/example/tasky/itemdetail/presentation/components/TimeDatePicker.kt

```diff
@@ -0,0 +1,173 @@
+package com.example.tasky.itemdetail.presentation.components
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ChevronRight
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.example.tasky.R
+import com.example.tasky.ui.theme.Black
+
+@Composable
+fun TimeDatePicker(
+    time: String,
+    timePrefix: String,
+    date: String,
+    isEditMode: Boolean,
+    onTimeClicked: () -> Unit,
+    onDateClicked: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Row(
+        modifier = modifier
+            .height(70.dp)
+            .fillMaxWidth()
+            .padding(horizontal = 16.dp),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.SpaceBetween
+    ) {
+
+        Row(
+            modifier = Modifier
+                .then(
+                    if (isEditMode) {
+                        Modifier.clickable { onTimeClicked() }
+                    } else {
+                        Modifier
+                    }
```

Comment:
AFAIK, the clickable modifier has an enabled property which you could set to `isEditMode`